src/DataMapper/ScheduleMapper.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GroupLife\Core\DataMapper;
+
+use GroupLife\Core\Schedule;
+
+class ScheduleMapper
+{
+    private $connection;
+    private $findSchedule;
+    private $insertSchedule;
+    private $insertScheduleRule;
+
+    public function __construct(\Doctrine\DBAL\Connection $connection)
+    {
+        $this->connection = $connection;
+
+        $this->insertSchedule = $this->connection->createQueryBuilder();
+        $this->insertSchedule
+            ->insert('schedule')->values(['id' => 'null']);
+
+        $this->insertScheduleRule = $this->connection->createQueryBuilder();
+        $this->insertScheduleRule
+            ->insert('schedule_rule')
+            ->values(['data' => '?', 'schedule' => '?', 'type' => '?']);
```

Comment:
This gives no benefit to create a query builder instance beforehand. When you need you may use `$db->insert()` directly. See https://www.doctrine-project.org/projects/doctrine-dbal/en/2.12/reference/data-retrieval-and-manipulation.html#insert